### DISCUSSION TITLE
Extension support for latest llama-index releases

### DISCUSSION THREAD

**Discussion by SLIBM** [IBM/ibm-generative-ai#329] [Q&A] [answered]
It seems that the official extension support for llama-index is on version 0.9.48, if I try `pip install 'ibm-generative-ai[llama-index]'` it will automatically install llama-index==0.9.48.
But this will not allow us to use new features offered by later versions of llama-index, wondering if there is a plan to upgrade the extension support for newer versions of llama-index?
Thanks!

**Comment by Tomas2D** [comment]
Hello!

You can install the newer version by running

```shell
pip install llama-index --upgrade
```

**Comment by SLIBM** [comment]
Actually I did try manual upgrading the llama-index lib, for example to the latest version of 0.10.14, I would get error from below snippet of code that used to run fine with llama-index 0.9.48 (this code snippet is the example code shared by documentation on how use the llama-index extension with ibm-generative-ai) 

```
from genai.extensions.llama_index import IBMGenAILlamaIndex
from genai.schema import DecodingMethod, TextGenerationParameters
index_llm = IBMGenAILlamaIndex(
    client=client,
    model_id="meta-llama/llama-2-70b-chat",
    parameters=TextGenerationParameters(
        decoding_method=DecodingMethod.SAMPLE,
        max_new_tokens=100,
        min_new_tokens=10,
        temperature=0.5,
        top_k=50,
        top_p=1,
    ),
)
```

and the error messages I got is like below:
```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
File ~/projects/RAG_BAM_EXP/venv/lib/python3.11/site-packages/genai/extensions/llama_index/embeddings.py:10
      9 try:
---> 10     from llama_index.embeddings.base import BaseEmbedding, Embedding
     11 except ImportError:

ModuleNotFoundError: No module named 'llama_index.embeddings.base'

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
Cell In[4], line 1
----> 1 from genai.extensions.llama_index import IBMGenAILlamaIndex
      2 from genai.schema import DecodingMethod, TextGenerationParameters
      3 index_llm = IBMGenAILlamaIndex(
      4     client=client,
      5     model_id="meta-llama/llama-2-70b-chat",
   (...)
     13     ),
     14 )

File ~/projects/RAG_BAM_EXP/venv/lib/python3.11/site-packages/genai/extensions/llama_index/__init__.py:3
      1 """Extension for LLamaIndex library"""
----> 3 from genai.extensions.llama_index.embeddings import IBMGenAILlamaIndexEmbedding
      4 from genai.extensions.llama_index.llm import IBMGenAILlamaIndex
      6 __all__ = ["IBMGenAILlamaIndex", "IBMGenAILlamaIndexEmbedding"]

File ~/projects/RAG_BAM_EXP/venv/lib/python3.11/site-packages/genai/extensions/llama_index/embeddings.py:12
     10     from llama_index.embeddings.base import BaseEmbedding, Embedding
     11 except ImportError:
---> 12     raise ImportError("Could not import llamaindex: Please install ibm-generative-ai[llama-index] extension.")  # noqa: B904
     15 class IBMGenAILlamaIndexEmbedding(BaseEmbedding):
     16     client: Client

ImportError: Could not import llamaindex: Please install ibm-generative-ai[llama-index] extension.
```

As you can see, the error is ImportError/ModuleNotFoundError originated from code out of the ibm genai extension code, like `~/projects/RAG_BAM_EXP/venv/lib/python3.11/site-packages/genai/extensions/llama_index/embeddings.py:10`

That's why I created this issue believing there may need some updates from ibm-generative-ai code on the extension support for llama-index to align with latest llama-index version.

**Reply by Tomas2D** [reply, marked as answer]
Ohh, I see. The fix is on the way #331

**Reply by SLIBM** [reply]
Thanks for the quick turnaround! I see it been merged, but there is no new release yet to pick it up

**Reply by Tomas2D** [reply]
We will release this week.

**Reply by Tomas2D** [reply]
Hopefully tomorrow.

**Reply by Tomas2D** [reply]
We are waiting for a new API deployment, for now. Feel free to use the `main` branch.

`pip install git+https://github.com/IBM/ibm-generative-ai.git@main`

**Reply by Tomas2D** [reply]
Released in version 2.3.0.
